app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -398,4 +399,54 @@ class AcraWrapper {
     fun putCustomString(key: String, value: String) {
         ACRA.getErrorReporter().putCustomData(key, value)
     }
+}
+
+class UsageUtility(private val prefs: SharedPreferences) {
+    private val numberOfTimesOpenedKey = "numberOfTimesOpened"
+    private val userGaveFeedbackKey = "userGaveFeedback"
+    private val dateTimeFirstOpenKey = "dateTimeFirstOpen"
+    private val millisecondsInThreeDays = 259200000L
+    private val minimumNumberOfOpensBeforeReviewRequest = 15
+
+    fun addToNumberOfTimesOpened(): Int {
+        val numberTimesOpened = prefs.getInt(numberOfTimesOpenedKey, 1)
+
+        if (numberTimesOpened == 1) {
+            with(prefs.edit()) {
+                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
+                putLong(dateTimeFirstOpenKey, Date().time)
+                apply()
+            }
+        } else {
+            with(prefs.edit()) {
+                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
+                apply()
+            }
+        }
+
+        return numberTimesOpened
+    }
+
+    fun getIsSufficientTimeElapsedSinceFirstOpen(): Boolean {
+        val dateTimeFirstOpened = prefs.getLong(dateTimeFirstOpenKey, 0L)
```

Comment:
I think you could combine `getIsSufficientTimeElapsedSinceFirstOpen`, `numberOfTimesOpenedIsGreaterThanThreshold`, and `getUserGaveFeedback` into one method, e.g. `UsageUtility#askingForFeedbackIsAppriopriate`.

Reply:
I made a function called `askingForFeedbackIsAppropriate` while hiding the other three as private functions.